core/state_transition.go

```diff
@@ -225,9 +218,9 @@ func (st *StateTransition) buyGas() error {
 	mgval = mgval.Mul(mgval, st.gasPrice)
 
 	if st.evm.ChainConfig().Scroll.FeeVaultEnabled() {
-		// always add l1fee, because all tx are L2-to-L1 ATM
-		log.Debug("Adding L1 fee", "l1_fee", st.l1Fee)
-		mgval = mgval.Add(mgval, st.l1Fee)
+		// always add l1DataFee, because all tx are L2-to-L1 ATM
```

Comment:
We should remove this comment or revise it because the `l1DataFee` for L1MessageTx should be 0

Reply:
fixed in https://github.com/scroll-tech/go-ethereum/pull/351/commits/9e223e9db85603a82aac0a859c28ffa4045cbfeb